aspnetcore/blazor/fundamentals/render-modes.md

```diff
@@ -0,0 +1,351 @@
+
+In the following example, the `Dialog` component doesn't specify a render mode. The render mode is applied where the `Dialog` component is used:
+
+<!-- UPDATE 8.0 RC2: Remove preview remark and update code -->
+
+```razor
+<Dialog @rendermode="new ServerRenderMode()" />
+```
+
+> [!NOTE]
+> During .NET 8 *Release Candidate 1*, use the following values:
+>
+> Render mode | Value
+> ----------- | -----
+> Server      | `new ServerRenderMode()`
+> WebAssembly | `new WebAssemblyRenderMode()`
+> Auto        | `new AutoRenderMode()`
+>
+> The preceding syntax will be simplified in an upcoming preview release.
+
+### Prerendering
+
+<!-- UPDATE 8.0 RC2: Remove preview remark and update code -->
+
+Prerendering is enabled by default for interactive components.
+
+To disable prerendering, pass the `prerender` flag with a value of `false`.
+
+For a routable page:
+
+```razor
+@page "{ROUTE}"
+@attribute [RenderModeServer(prerender: false)]
+```
+
+In the preceding example, the `{ROUTE}` placeholder is the route template.
+
+For a non-routable, non-page component:
+
+```razor
+<Dialog @rendermode="new ServerRenderMode(prerender: false)" />
+```
+
+> [!NOTE]
+> During .NET 8 *Release Candidate 1*, use the following values:
+>
+> Render mode | Value
+> ----------- | -----
+> Server      | `new ServerRenderMode(prerender: false)`
+> WebAssembly | `new WebAssemblyRenderMode(prerender: false)`
+> Auto        | `new AutoRenderMode(prerender: false)`
+>
+> The preceding syntax will be simplified in an upcoming preview release.
+
+### Static render mode
+
+By default components use the Static render mode. The component renders to the response stream and interactivity isn't enabled.
+
+In the following example, there's no designation for the component's render mode. Therefore, the component is *statically rendered* on the server. The button isn't interactive and doesn't call the `UpdateMessage` method when selected. The value of `message` doesn't change, and the component isn't rerendered in response to UI events.
+
+`RenderMode1.razor`:
+
+```razor
+@page "/render-mode-1"
+
+<button @onclick="UpdateMessage">Click me</button> @message
+
+@code {
+    private string message = "Not clicked yet.";
+
+    private void UpdateMessage()
+    {
+        message = "Somebody clicked me!";
+    }
+}
+```
+
+If using the preceding component locally in a Blazor Web App, place the component in the server-side project's `Components/Pages` folder. The server-side project is the solution's project with a name that doesn't end in `.Client`. When the app is running, navigate to `/render-mode-1` in the browser's address bar.
+
+> [!NOTE]
+> The anatomy of a basic Razor component is fully explained in the <xref:blazor/fundamentals/index> article. Project structure for apps created by a Blazor template are described in the <xref:blazor/project-structure> article. Detailed components coverage is found in the *Components* articles later in the documentation.
```

Comment:
Do we need to repeat these instructions in every section? It distracts from the main content. There are instructions at the top of the doc on how to run and test the samples locally.

Reply:
No ... and it could be removed. This is a bit of a holdover from when I had the *Render modes* topic sitting directly under the *Hosting Models* topic in the ToC. That wasn't so good because it made the components in this topic the first ones that devs saw.

It's still an open question if this whole topic should go between the *Fundamenals* and *Components* node. That might be ok now because we won't show a render mode in (or most of) the component examples in the *Fundamentals* node after RC2.

Reply:
..... or go back to placing it under the *Hosting Models* article ..... or place it as the first topic in the *Components* node under the overview there. I'm not sure what the absolute best spot is. It's certainly more flexible now that render modes don't need to be explained for early examples where it would be present without explanation (after RC2).

Reply:
I just removed that NOTE.

Reply:
> or go back to placing it under the Hosting Models article

By "under" do you mean after or nested?

Reply:
I meant after the *Hosting Models* article, but that location still suffers a bit from readers not learning more about components first.

The main reason that I moved it was to get it immediately after where components are first explained. They're first explained in the *Fundamentals* overview because other *Fundamentals* node topics have component examples in them. However, the *Fundamentals* node is full of app-level concerns ... routing, logging, error handling, etc. Render modes seems more component-focused. How about as the first article after the *Components* node overview? That's at least a better option now that I know we won't (after RC2) be placing a render mode at the top of every page component example.

Reply:
> How about as the first article after the Components node overview

That sounds fine. I'm pretty flexible on where the topic goes in the TOC. as long as it's in a reasonable place in the hierarchy. Right now, I think the Blazor docs are more hierarchical than sequential.

Reply:
Ok, sounds good.

Reply:
> hierarchical than sequential.

I know that readers read them straight down the ToC. Many have told me so over the years. It's logical, too, coming into the whole framework knowing nothing about it.